src/drivers/imx/sdma.c

```diff
@@ -773,7 +773,7 @@ static int sdma_prep_desc(struct dma_chan_data *channel,
 		return -EINVAL;
 	}
 
-	watermark = config->burst_elems;
+	watermark = (config->burst_elems * 4) / 8;
```

Comment:
Where does each of multiplying by 4 and dividing by 8 come from? I feel like at least the multiplication by 4 can be replaced with a multiplication by the width?

Reply:
@paulstelian97 yes 4 comes from width and i will replace it. Then / 8 comes from multiple run tests, where this was the best option.

Ideally, as SDMA owner told me it should work using the same watermark as the DAI.  I will still investigate this until we completely understand what happens. 

So, lets just keep this PR open for few more days.

Reply:
The main thing is, the SDMA watermark must divide the size of the BD otherwise we get unexpected behavior that is harder to quantify. Does this ensure that?

Reply:
@paulstelian97 yes, that's ensure that. And I choose this values explicitly to validate the condition. The thing is that the documentation doesn't explicitly mention about watermark to divide the BD count :).

Reply:
Well it technically doesn't _break_ if you don't ensure the condition, but the behavior we're getting is weird because at the end of the BD it could definitely transfer less than what the watermark says.

Reply:
Indeed, these indeed are fine tuning patches. That we will surely do in the future. But lets now have an SDMA driver which is stable and gets work done.

Reply:
`width`?

Reply:
oh, thanks for noticing that. I thought that Ive already fixed that. Should be ok now.